src/Tribe/RSVP.php

```diff
@@ -219,6 +219,109 @@ public function hooks() {
+		add_action( 'wp_ajax_tribe_tickets_rsvp_handle', [ $this, 'ajax_handle_rsvp' ] );
+	}
+
+	/**
+	 * Handle RSVP processing for the RSVP forms.
+	 *
+	 * @since TBD
+	 */
+	public function ajax_handle_rsvp() {
+		$response = [
+			'html' => '',
+		];
+
+		$post_id   = absint( tribe_get_request_var( 'post_id', 0 ) );
+		$ticket_id = absint( tribe_get_request_var( 'ticket_id', 0 ) );
+		$step      = tribe_get_request_var( 'step', null );
+
+		$html = $this->render_rsvp_step( $ticket_id, $post_id, $step );
+
+		if ( '' === $html ) {
+			return wp_send_json_error( $response );
+		}
+
+		$response['html'] = $html;
+
+		return wp_send_json_success( $response );
+	}
+
+	/**
+	 * Handle RSVP processing for the RSVP forms.
+	 *
+	 * @since TBD
+	 *
+	 * @param int         $ticket_id The ticket ID.
+	 * @param int         $post_id   The post or event ID.
+	 * @param null|string $step      Which step to render.
+	 *
+	 * @return string The step template HTML.
+	 */
+	public function render_rsvp_step( $ticket_id, $post_id, $step = null ) {
+		// No ticket / post ID.
+		if ( 0 === $post_id || 0 === $ticket_id ) {
+			return '';
+		}
+
+		/** @var Tribe__Tickets__RSVP $rsvp */
+		$rsvp = tribe( 'tickets.rsvp' );
+
+		/** @var \Tribe__Tickets__Editor__Blocks__Rsvp $blocks_rsvp */
+		$blocks_rsvp = tribe( 'tickets.editor.blocks.rsvp' );
+
+		/** @var \Tribe__Tickets__Editor__Template $template */
+		$template = tribe( 'tickets.editor.template' );
+
+		$ticket = $rsvp->get_ticket( $post_id, $ticket_id );
+
+		// No ticket found.
+		if ( null === $ticket ) {
+			return '';
+		}
+
+		// Set required template globals.
+		$args = [
+			'rsvp_id'    => $ticket_id,
+			'post_id'    => $post_id,
+			'rsvp'       => $ticket,
+			'step'       => $step,
+			'must_login' => ! is_user_logged_in() && $rsvp->login_required(),
+			'login_url'  => Tribe__Tickets__Tickets::get_login_url( $post_id ),
+			'threshold'  => $blocks_rsvp->get_threshold( $post_id ),
+		];
+
+		/**
+		 * Handle processing for the RSVP step if necessary.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $args {
+		 *      The list of step template arguments.
+		 *
+		 *      @type int                           $rsvp_id    The RSVP ticket ID.
+		 *      @type int                           $post_id    The ticket ID.
+		 *      @type Tribe__Tickets__Ticket_Object $rsvp       The RSVP ticket object.
+		 *      @type null|string                   $step       Which step being rendered.
+		 *      @type boolean                       $must_login Whether login is required to register.
+		 *      @type string                        $login_url  The site login URL.
+		 *      @type int                           $threshold  The RSVP ticket threshold.
+		 * }
+		 */
+		do_action( 'tribe_tickets_rsvp_process_rsvp_step', $args );
+
+		// Add the rendering attributes into global context.
+		$template->add_template_globals( $args );
+
+		$html = $template->template( 'v2/components/loader/loader', [], false );
```

Comment:
:warning: Equals sign not aligned with surrounding assignments; expected 2 spaces but found 1 space

<sub>hash: 6c71292d467b4868c689bbbc4889bf62</sub>